tests/accelerators/test_dp.py

```diff
@@ -95,3 +98,53 @@ def test_dp_test(tmpdir):
     new_weights = model.c_d1.weight.clone().detach().cpu()
 
     assert torch.all(torch.eq(old_weights, new_weights))
+
+
+@pytest.mark.skipif(torch.cuda.device_count() < 2, reason="test requires multi-GPU machine")
+def test_dp_raise_exception_with_batch_transfer_hooks(tmpdir):
+    """
+    Test that an exception is raised when overriding batch_transfer_hooks in DP model.
+    """
+
+    class CustomModel(BoringModel):
+
+        def transfer_batch_to_device(self, batch, device):
+            batch = batch.cuda()
+            return batch
+
+        def transform_hook(self, batch, dataloader_idx):
+            batch += 1
+            return batch
+
+    tutils.set_random_master_port()
+    trainer_options = dict(
+        default_root_dir=tmpdir,
+        max_steps=7,
+        gpus=[0, 1],
+        accelerator='dp',
+    )
+
+    # Override transfer_batch_to_device hook only
+    trainer = Trainer(**trainer_options)
+    model = CustomModel()
+
+    with pytest.raises(MisconfigurationException, match=r'Overriding `transfer_batch_to_device` is not .* in DP'):
+        trainer.fit(model)
+
+    # Override on_before_batch_transfer hook only
+    trainer = Trainer(**trainer_options)
+    model = CustomModel()
+    model.transfer_batch_to_device = BoringModel().transfer_batch_to_device
+    model.on_before_batch_transfer = model.transform_hook
```

Comment:
Can you avoid patching by just instantiating boring model or proper subclassing?